### PR TITLE
[ecobee] Remove org.apache.commons

### PR DESCRIPTION
**Pull request by lsiepel**
- Remove dependency on apache.commons

Untested, minor refactoring from only from code perspective

### CODE REVIEW

**Review by mhilbush — APPROVED** [review]
Thanks for doing this!

These changes all LGTM.

It's a bit unfortunate that there's no common ExceptionUtils class in core. I would imagine this binding isn't the only one that needs to get the root cause exception.

Same for the utility to capitalize words.

File: bundles/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/util/StringUtils.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * The {@link StringUtils} class defines static string related methods
+ *
+ * @author Leo Siepel - Initial contribution
+ */
+@NonNullByDefault
+public class StringUtils {
+
+    public static @Nullable String capitalizeWords(@Nullable String input) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/util/StringUtils.java:26]: Looks like the method can't return a null value.

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/util/StringUtils.java (hunk 1 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/util/StringUtils.java:26]: Fixed

**Review by lolodomo — APPROVED** [review]
LGTM

**Pull request merged**